.github/workflows/publish-internal.yml

```diff
@@ -70,10 +73,13 @@ jobs:
         with:
           version_number: ${{ steps.package.outputs.version }}
           versions_published: ${{ steps.published.outputs.versions }}
-      - run: |
-          hatch version ${{ steps.next.outputs.internal_release_version }}+$(git rev-parse HEAD)
-          hatch build --clean
-          hatch run build:check-all
+      - name: "Update version to internal PyPI format"
```

Comment:
It turns out `hatch version <VERSION>` will coerce `VERSION` into a `PEP440` compliant version. For example, `hatch version 1.9.0post1+abcd` will make the version `1.9.0.post1+abcd`. However, we verify that the version of the package is a `SemVer` compliant version (which `1.9.0.post1+abcd` is not). In this example, we're expecting `1.9.0post1+abcd`.

What made this worse is that CodeArtifact will also coerce the version before displaying it in their UI. So it is not obvious that the file contains `1.9.0post1+abcd` because it displays `1.9.0.post1+abcd`.

We need to replace the version manually instead of using standard build tools as a result.